### ISSUE TITLE
你好，请问下你是怎么导出NCNN模型的？

### ISSUE THREAD

**Issue by ashuowu** [mpj1234/ncnn-yolov12-android#1]
你的项目构建使用成功，但是我自己转换成NCNN模型会失败，能告知你是如何导出NCNN的吗

**Comment by mpj1234** [comment]
导出模型时候去掉batch维度，因为ncnn不支持

**Comment by ashuowu** [comment]
好的，非常感谢您！我这就去尝试。

**Comment by ashuowu** [comment]
> 导出模型时候去掉batch维度，因为ncnn不支持

去掉batch维度之后出现layer ReduceMax not exists or registered这个错误，老哥方便的话能不能出个教程之类的呢，谢谢了。

**Comment by mpj1234** [comment]
> > 导出模型时候去掉batch维度，因为ncnn不支持
> 
> 去掉batch维度之后出现layer ReduceMax not exists or registered这个错误，老哥方便的话能不能出个教程之类的呢，谢谢了。

最近有大项目在做，得等时间了

**Comment by Kk875** [comment]
> > 导出模型时候去掉batch维度，因为ncnn不支持
> 
> 去掉batch维度之后出现layer ReduceMax not exists or registered这个错误，老哥方便的话能不能出个教程之类的呢，谢谢了。

请问解决了吗？老铁

**Comment by ashuowu** [comment]
> > > 导出模型时候去掉batch维度，因为ncnn不支持
> > 
> > 
> > 去掉batch维度之后出现layer ReduceMax not exists or registered这个错误，老哥方便的话能不能出个教程之类的呢，谢谢了。
> 
> 请问解决了吗？老铁
没有呢

**Comment by mpj1234** [comment]
请查看readme

**Comment by Kk875** [comment]
> 请查看readme

----------------

############# pass_level1
############# pass_level2
############# pass_level3
############# pass_level4
############# pass_level5
############# pass_ncnn
fallback batch axis 233 for operand 40
fallback batch axis 233 for operand 41
fallback batch axis 233 for operand 42
fallback batch axis 233 for operand 68
fallback batch axis 233 for operand 69
fallback batch axis 233 for operand 70
fallback batch axis 233 for operand 96
fallback batch axis 233 for operand 97
fallback batch axis 233 for operand 98
fallback batch axis 233 for operand 124
fallback batch axis 233 for operand 125
fallback batch axis 233 for operand 126
NCNN: export success ✅ 1.2s, saved as 'runs/detect/train/weights/best_ncnn_model' (9.8 MB)

Export complete (2.4s)
Results saved to /home/user/Project/yolov12-main/runs/detect/train/weights
Predict:         yolo predict task=detect model=runs/detect/train/weights/best_ncnn_model imgsz=640  
Validate:        yolo val task=detect model=runs/detect/train/weights/best_ncnn_model imgsz=640 data=vehicles.yaml  
Visualize:       https://netron.app

导出时还是会报错 fallback batch axis 233 for operand 40

**Comment by mpj1234** [comment]
是否使用最新的pnnx，必须要最新的，旧版pnnx需要修改模型源码

**Comment by Kk875** [comment]
> 是否使用最新的pnnx，必须要最新的，旧版pnnx需要修改模型源码

最新版的也无法导出：我使用的版本是pnnx-20250430-linux
----------------

############# pass_level1
############# pass_level2
############# pass_level3
############# pass_level4
############# pass_level5
############# pass_ncnn
fallback batch axis 233 for operand 40
fallback batch axis 233 for operand 41
fallback batch axis 233 for operand 42
fallback batch axis 233 for operand 43
fallback batch axis 233 for operand 44
fallback batch axis 233 for operand 45
fallback batch axis 233 for operand 46
fallback batch axis 233 for operand 47
fallback batch axis 233 for operand 49
fallback batch axis 233 for operand 51
fallback batch axis 233 for operand 53
fallback batch axis 233 for operand 54
fallback batch axis 233 for operand 68
fallback batch axis 233 for operand 69
fallback batch axis 233 for operand 70
fallback batch axis 233 for operand 71
fallback batch axis 233 for operand 72
fallback batch axis 233 for operand 73
fallback batch axis 233 for operand 74
fallback batch axis 233 for operand 75
fallback batch axis 233 for operand 77
fallback batch axis 233 for operand 79
fallback batch axis 233 for operand 81
fallback batch axis 233 for operand 82
fallback batch axis 233 for operand 96
fallback batch axis 233 for operand 97
fallback batch axis 233 for operand 98
fallback batch axis 233 for operand 99
fallback batch axis 233 for operand 100
fallback batch axis 233 for operand 101
fallback batch axis 233 for operand 102
fallback batch axis 233 for operand 103
fallback batch axis 233 for operand 105
fallback batch axis 233 for operand 107
fallback batch axis 233 for operand 109
fallback batch axis 233 for operand 110
fallback batch axis 233 for operand 124
fallback batch axis 233 for operand 125
fallback batch axis 233 for operand 126
fallback batch axis 233 for operand 127
fallback batch axis 233 for operand 128
fallback batch axis 233 for operand 129
fallback batch axis 233 for operand 130
fallback batch axis 233 for operand 131
fallback batch axis 233 for operand 133
fallback batch axis 233 for operand 135
fallback batch axis 233 for operand 137
fallback batch axis 233 for operand 138
fallback batch axis 233 for operand pnnx_expr_532_mul(47,0.176777)
fallback batch axis 233 for operand pnnx_expr_528_sub(pnnx_expr_532_mul(47,0.176777),49)
fallback batch axis 233 for operand pnnx_expr_528_exp(sub(pnnx_expr_532_mul(47,0.176777),49))
fallback batch axis 233 for operand pnnx_expr_526_div(pnnx_expr_528_exp(sub(pnnx_expr_532_mul(47,0.176777),49)),51)
fallback batch axis 233 for operand pnnx_expr_447_mul(75,0.176777)
fallback batch axis 233 for operand pnnx_expr_443_sub(pnnx_expr_447_mul(75,0.176777),77)
fallback batch axis 233 for operand pnnx_expr_443_exp(sub(pnnx_expr_447_mul(75,0.176777),77))
fallback batch axis 233 for operand pnnx_expr_441_div(pnnx_expr_443_exp(sub(pnnx_expr_447_mul(75,0.176777),77)),79)
fallback batch axis 233 for operand pnnx_expr_362_mul(103,0.176777)
fallback batch axis 233 for operand pnnx_expr_358_sub(pnnx_expr_362_mul(103,0.176777),105)
fallback batch axis 233 for operand pnnx_expr_358_exp(sub(pnnx_expr_362_mul(103,0.176777),105))
fallback batch axis 233 for operand pnnx_expr_356_div(pnnx_expr_358_exp(sub(pnnx_expr_362_mul(103,0.176777),105)),107)
fallback batch axis 233 for operand pnnx_expr_277_mul(131,0.176777)
fallback batch axis 233 for operand pnnx_expr_273_sub(pnnx_expr_277_mul(131,0.176777),133)
fallback batch axis 233 for operand pnnx_expr_273_exp(sub(pnnx_expr_277_mul(131,0.176777),133))
fallback batch axis 233 for operand pnnx_expr_271_div(pnnx_expr_273_exp(sub(pnnx_expr_277_mul(131,0.176777),133)),135)
NCNN: export success ✅ 1.2s, saved as 'runs/detect/train/weights/best_ncnn_model' (9.8 MB)

**Comment by mpj1234** [comment]
将pt模型发出来

**Comment by mpj1234** [comment]
```
D:\Anaconda\install\envs\llm\python.exe F:\python\yolov12-main\export.py 
FlashAttention is not available on this device. Using scaled_dot_product_attention instead.
D:\Anaconda\install\envs\llm\lib\site-packages\timm\models\layers\__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
Ultralytics 8.3.63 🚀 Python-3.10.14 torch-2.3.1+cu118 CPU (12th Gen Intel Core(TM) i5-12600K)
YOLOv12n summary (fused): 352 layers, 2,590,824 parameters, 0 gradients, 6.5 GFLOPs

PyTorch: starting from 'weights\yolov12n.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 84, 8400) (5.3 MB)

TorchScript: starting export with torch 2.3.1+cu118...
TorchScript: export success ✅ 2.1s, saved as 'weights\yolov12n.torchscript' (10.6 MB)

NCNN: starting export with NCNN 1.0.20241226...
NCNN: running 'F:\python\yolov12-main\ultralytics\pnnx.exe weights\yolov12n.torchscript ncnnparam=weights\yolov12n_ncnn_model\model.ncnn.param ncnnbin=weights\yolov12n_ncnn_model\model.ncnn.bin ncnnpy=weights\yolov12n_ncnn_model\model_ncnn.py pnnxparam=weights\yolov12n_ncnn_model\model.pnnx.param pnnxbin=weights\yolov12n_ncnn_model\model.pnnx.bin pnnxpy=weights\yolov12n_ncnn_model\model_pnnx.py pnnxonnx=weights\yolov12n_ncnn_model\model.pnnx.onnx fp16=0 device=cpu inputshape="[1, 3, 640, 640]"'
pnnxparam = weights\yolov12n_ncnn_model\model.pnnx.param
pnnxbin = weights\yolov12n_ncnn_model\model.pnnx.bin
pnnxpy = weights\yolov12n_ncnn_model\model_pnnx.py
pnnxonnx = weights\yolov12n_ncnn_model\model.pnnx.onnx
ncnnparam = weights\yolov12n_ncnn_model\model.ncnn.param
ncnnbin = weights\yolov12n_ncnn_model\model.ncnn.bin
ncnnpy = weights\yolov12n_ncnn_model\model_ncnn.py
fp16 = 0
optlevel = 2
device = cpu
inputshape = [1,3,640,640]f32
inputshape2 = 
customop = 
moduleop = 
get inputshape from traced inputs
inputshape = [1,3,640,640]f32
############# pass_level0
inline module = torch.nn.modules.linear.Identity
inline module = ultralytics.nn.modules.block.A2C2f
inline module = ultralytics.nn.modules.block.AAttn
inline module = ultralytics.nn.modules.block.ABlock
inline module = ultralytics.nn.modules.block.Bottleneck
inline module = ultralytics.nn.modules.block.C3k
inline module = ultralytics.nn.modules.block.C3k2
inline module = ultralytics.nn.modules.block.DFL
inline module = ultralytics.nn.modules.conv.Concat
inline module = ultralytics.nn.modules.conv.Conv
inline module = ultralytics.nn.modules.conv.DWConv
inline module = ultralytics.nn.modules.head.Detect
inline module = torch.nn.modules.linear.Identity
inline module = ultralytics.nn.modules.block.A2C2f
inline module = ultralytics.nn.modules.block.AAttn
inline module = ultralytics.nn.modules.block.ABlock
inline module = ultralytics.nn.modules.block.Bottleneck
inline module = ultralytics.nn.modules.block.C3k
inline module = ultralytics.nn.modules.block.C3k2
inline module = ultralytics.nn.modules.block.DFL
inline module = ultralytics.nn.modules.conv.Concat
inline module = ultralytics.nn.modules.conv.Conv
inline module = ultralytics.nn.modules.conv.DWConv
inline module = ultralytics.nn.modules.head.Detect

----------------

############# pass_level1
############# pass_level2
############# pass_level3
############# pass_level4
############# pass_level5
############# pass_ncnn
NCNN: export success ✅ 1.9s, saved as 'weights\yolov12n_ncnn_model' (10.1 MB)

Export complete (4.6s)
Results saved to F:\python\yolov12-main\weights
Predict:         yolo predict task=detect model=weights\yolov12n_ncnn_model imgsz=640  
Validate:        yolo val task=detect model=weights\yolov12n_ncnn_model imgsz=640 data=None  
Visualize:       https://netron.app

进程已结束，退出代码为 0
```
这是我的日志
